gobblin-compaction/src/main/java/gobblin/compaction/dataset/CompactionSuite.java

```diff
@@ -0,0 +1,42 @@
+package gobblin.compaction.dataset;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.compaction.verify.CompactionAuditCountVerifier;
+import gobblin.compaction.verify.CompactionThresholdVerifier;
+import gobblin.compaction.verify.CompactionVerifier;
+import gobblin.configuration.State;
+
+/**
+ * This class represents a sort of utilities {@link CompactionPartition} will use to parse or verify the data.
+ *
+ * A {@link CompactionParser} is provided
+ * A map of {@link CompactionVerifier} is provided
+ */
+@Slf4j
+@AllArgsConstructor
+public class CompactionSuite {
```

Comment:
Consider having `CompactionSuite` as the place where all compaction business logic is located. If that is the case, `CompactionTask` is just a bridge between a `Dataset`, a `CompactionSuite`, and the `MRTask`.